modules/framework/galasa-parent/dev.galasa.framework.api.monitors/src/main/java/dev/galasa/framework/api/monitors/internal/MonitorTransform.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.api.monitors.internal;
+
+import java.util.List;
+
+import dev.galasa.framework.api.beans.generated.GalasaMonitor;
+import dev.galasa.framework.api.beans.generated.GalasaMonitordata;
+import dev.galasa.framework.api.beans.generated.GalasaMonitordataResourceCleanupData;
+import dev.galasa.framework.api.beans.generated.GalasaMonitordataresourceCleanupDatafilters;
+import dev.galasa.framework.api.beans.generated.GalasaMonitormetadata;
+import dev.galasa.framework.api.common.resources.GalasaResourceValidator;
+import io.kubernetes.client.openapi.models.V1Container;
+import io.kubernetes.client.openapi.models.V1Deployment;
+import io.kubernetes.client.openapi.models.V1DeploymentSpec;
+import io.kubernetes.client.openapi.models.V1EnvVar;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+
+/**
+ * Converts a Kubernetes deployment into an external GalasaMonitor bean.
+ */
```

Comment:
Might be helpful to document an example deployment and which bits we are interested in for the bean ? Not sure. It's hard to see from the code what the yaml might have looked like.

Reply:
Addressed in https://github.com/galasa-dev/galasa/pull/237/commits/f451d8dd1e34e1529fc9e157d1170d93284a8066